src/main/scala/com/softwaremill/realworld/articles/ArticlesService.scala

```diff
@@ -0,0 +1,38 @@
+package com.softwaremill.realworld.articles
+
+import com.softwaremill.realworld.profiles.ProfilesRepository
+import zio.{IO, ZIO, ZLayer}
+
+class ArticlesService(articlesRepository: ArticlesRepository, profilesRepository: ProfilesRepository):
+
+  def list(): IO[String, List[Article]] = articlesRepository
+    .list()
+    .flatMap(sas => ZIO.collectAll(sas.map(article)))
+    .onError(err => ZIO.logError(err.prettyPrint))
+
+  private def article(a: StoredArticle): IO[String, Article] = {
+    profilesRepository
+      .find(a.authorId)
```

Comment:
isn't that inefficient, where we query for an author each time? I guess a dedicated SQL query would be best here, but maybe that's coming in the next PR :)

Reply:
Yep, sql is coming soon :)